File: core/src/com/sheepsgohome/leaderboard/LeaderBoardCallback.java
package com.sheepsgohome.leaderboard;

public interface LeaderBoardCallback {

    void connecting();

    void connectionToDatabaseFailed();

    void invalidData();

    void nickAlreadyInUse();

    void success();

    void failure();

    void failedToInitializeMD5();

    void connectionFailed();

    void connectionCanceled();

    void unregisteredUser();

    void leaderboardResult(LeaderBoardResult result);
}

File: core/src/com/sheepsgohome/leaderboard/LeaderBoardCallback.kt
package com.sheepsgohome.leaderboard

interface LeaderBoardCallback {

    fun connecting()

    fun connectionToDatabaseFailed()

    fun invalidData()

    fun nickAlreadyInUse()

    fun success()

    fun failure()

    fun failedToInitializeMD5()

    fun connectionFailed()

    fun connectionCanceled()

    fun unregisteredUser()

    fun leaderboardResult(result: LeaderBoardResult)
}
